modules/Forum/module.php

```diff
@@ -120,14 +120,43 @@ public function __construct(Language $language, Language $forum_language, Pages
             true
         );
 
+        EventHandler::registerEvent('renderPost',
+            $this->_forum_language->get('forum', 'render_post'),
+            [
+                'content' => $this->_language->get('general', 'content')
+            ],
+            true,
+            true
+        );
+
+        EventHandler::registerEvent('renderPostEdit',
+            $this->_forum_language->get('forum', 'render_post_edit'),
+            [
+                'content' => $this->_language->get('general', 'content')
+            ],
+            true,
+            true
+        );
+
         require_once ROOT_PATH . '/modules/Forum/hooks/DeleteUserForumHook.php';
         EventHandler::registerListener('deleteUser', 'DeleteUserForumHook::execute');
 
+        require_once ROOT_PATH . '/modules/Core/hooks/ContentHook.php';
         require_once ROOT_PATH . '/modules/Forum/hooks/MentionsHook.php';
         EventHandler::registerListener('prePostCreate', 'MentionsHook::preCreate');
         EventHandler::registerListener('prePostEdit', 'MentionsHook::preEdit');
         EventHandler::registerListener('preTopicCreate', 'MentionsHook::preCreate');
         EventHandler::registerListener('preTopicEdit', 'MentionsHook::preEdit');
+        EventHandler::registerListener('renderPost', 'ContentHook::purify');
```

Comment:
Random thought, all of these pre/post render "events" aren't exactly the same type of event as the other usage of the eventhandler class. 

To avoid confusion and separate them, could we create something like a "pipeline" class? Basically the same thing, but used for """under the hood""" events instead of normal events such as user register/verify etc.

Reply:
When you register an event, it's possible to mark it as internal which distinguishes the event from ones the user should be aware of: https://github.com/NamelessMC/Nameless/blob/v2/core/classes/Events/EventHandler.php#L40